three-dna/src/index.js

```
@@ -0,0 +1,599 @@
+'use strict';
+
+/*
+ThreeDna 0.0.1
+license: MIT
+https://github.com/MichaelMass/three-science
+*/
+class ThreeDna {
+  constructor(opts) {
+    opts = !!opts ? opts : {};
+
+    this._params = {};
+    this._opts = {};
+    this._view = {};
+    this._obj = {};
+    this._obj.mat = {};
+    this._obj.geo = {};
+
+    this._container = opts.container && ThreeDna.Utils.isElement(opts.container) ? opts.container : ThreeDna.Utils._createContainer();
```

It's as you please but I think that the only thing that should be mandatory to pass as an argument is the container element ... in this way you wouldn't need to implement all these helper functions ... You could throw an Error if the element is not an element instead of creating one for the user ..